### PR TITLE
Tf 131 update

### PR DESCRIPTION
**Pull request by marcreichman**
Update to tensorflow_cc default distribution (which updates to a newer TF release that can handle CUDA 10) and integrate its version of a CPU-only build.

Created as a draft while docker builds are tested

### CODE REVIEW

**Comment by beniz** [comment]
HI @marcreichman , thanks for the PR, very much appreciated. Our current automatic CI from github repo changes doesn't test TF backend (see https://github.com/jolibrain/deepdetect/blob/master/Jenkinsfile). Let us know if you need us to replicate some builds.

Also, FYI, there's a TF+DD image classification unit test (https://github.com/jolibrain/deepdetect/blob/master/tests/ut-tfapi.cc).

**Comment by beniz** [comment]
Hi, FYI the build goes through for us with bazel 0.24.1, but the tests `ut_tf_api` do fail with a protobuf error that may indicate an incompatibility when reading the models used in the test.

**Comment by marcreichman** [comment]
Hi @beniz, for what it's worth, we've been testing a variation of this build (our own custom docker build, templated from yours) with our pre-existing TF models and there hasn't been an issue. Which models are you using that are failing?

**Comment by beniz** [comment]
Hello, thanks for the update. We're using https://www.deepdetect.com/models/tf/inception_v1.pb and other models are available from https://www.deepdetect.com/models/tf/. We could definitely modify our tests to use another public model that works fine.

Is `./ut_tf_api` passing in your docker build ?

**Comment by marcreichman** [comment]
So, in our test build `ut_tfapi` is running fine:

```root@58db32b561fe:/opt/deepdetect/build/tests# ./ut_tfapi
Running main() from gtest_main.cc
[==========] Running 2 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 2 tests from tfapi
[ RUN      ] tfapi.service_predict
[2019-11-14 19:13:18.236] [imgserv] [info] predict: using graphFile dir=../examples/tf/inception//inception_v1.pb
[2019-11-14 19:13:18.398] [imgserv] [info] using input layer=InputImage
[2019-11-14 19:13:18.398] [imgserv] [info] using output layer=InceptionV1/Logits/Predictions/Reshape_1
2019-11-14 19:13:18.398774: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX AVX2 FMA
2019-11-14 19:13:18.435758: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 1700020000 Hz
2019-11-14 19:13:18.436530: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x5601a4febc30 executing computations on platform Host. Devices:
2019-11-14 19:13:18.436587: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
joutstr={"status":{"code":200,"msg":"OK"},"head":{"method":"/predict","service":"imgserv","time":1333.0},"body":{"predictions":[{"classes":[{"prob":0.40899214148521426,"cat":"n03763968 military uniform"},{"prob":0.047941841185092929,"cat":"n04350905 suit, suit of clothes"},{"last":true,"cat":"n02916936 bulletproof vest","prob":0.032782118767499927}],"uri":"../examples/tf/inception/grace_hopper.jpg"}]}}
joutstr={"status":{"code":200,"msg":"OK"},"head":{"method":"/predict","service":"imgserv","time":158.0},"body":{"predictions":[{"classes":[{"prob":0.40899229049682619,"cat":"n03763968 military uniform"},{"prob":0.04794185981154442,"cat":"n04350905 suit, suit of clothes"},{"last":true,"cat":"n02916936 bulletproof vest","prob":0.03278209641575813}],"uri":"../examples/tf/inception/grace_hopper.jpg"},{"classes":[{"prob":0.5209034085273743,"cat":"n02123045 tabby, tabby cat"},{"prob":0.2193935662508011,"cat":"n02124075 Egyptian cat"},{"last":true,"cat":"n02123159 tiger cat","prob":0.11031948029994965}],"uri":"../examples/tf/inception/cat.jpg"}]}}
[       OK ] tfapi.service_predict (1600 ms)
[ RUN      ] tfapi.service_predict_unsup
[2019-11-14 19:13:19.656] [imgserv] [info] predict: using graphFile dir=../examples/tf/inception//inception_v1.pb
[2019-11-14 19:13:19.669] [imgserv] [info] using input layer=InputImage
[       OK ] tfapi.service_predict_unsup (741 ms)
[----------] 2 tests from tfapi (2342 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test case ran. (2343 ms total)
[  PASSED  ] 2 tests.```


We have been testing this on 18.04 for other changes (I work with @cchadowitz-pf), and while I've had to do a number of small package updates, for the most part things have built ok, including our own integration tests with the models we're using. It took me a while to get the tests going in the docker environment. When I do, one test does fail, `ut_conn` in the `outputconn.cmfull` test:

```
[ RUN      ] outputconn.cmfull
jstr={"measure":{"cmfull":[{"zero":[0.5,0.5,0.0,0.0]},{"one":[0.0,1.0,0.0,0.0]},{"two":[0.0,1.0,0.0,0.0]},{"three":[2.696539702293474e308,2.696539702293474e308,2.696539702293474e308,2.696539702293474e308]}],"precision":0.3749999968750001,"labels":["zero","one","two","three"],"f1":0.35294117352941187,"accp":0.5,"recall":0.3333333305555556,"cmdiag":[0.4999999975,0.9999999900000002,0.0,0.0]}}
/opt/deepdetect/tests/ut-conn.cc:285: Failure
      Expected: "{\"measure\":{\"labels\":[\"zero\",\"one\",\"two\",\"three\"],\"f1\":0.35294117352941187,\"cmfull\":[{\"zero\":[0.5,0.5,0.0,0.0]},{\"one\":[0.0,1.0,0.0,0.0]},{\"two\":[0.0,1.0,0.0,0.0]},{\"three\":[2.696539702293474e308,2.696539702293474e308,2.696539702293474e308,2.696539702293474e308]}],\"cmdiag\":[0.4999999975,0.9999999900000002,0.0,0.0],\"recall\":0.3333333305555556,\"precision\":0.3749999968750001,\"accp\":0.5}}"
To be equal to: jstr
      Which is: "{\"measure\":{\"cmfull\":[{\"zero\":[0.5,0.5,0.0,0.0]},{\"one\":[0.0,1.0,0.0,0.0]},{\"two\":[0.0,1.0,0.0,0.0]},{\"three\":[2.696539702293474e308,2.696539702293474e308,2.696539702293474e308,2.696539702293474e308]}],\"precision\":0.3749999968750001,\"labels\":[\"zero\",\"one\",\"two\",\"three\"],\"f1\":0.35294117352941187,\"accp\":0.5,\"recall\":0.3333333305555556,\"cmdiag\":[0.4999999975,0.9999999900000002,0.0,0.0]}}"
[  FAILED  ] outputconn.cmfull (3 ms)
```

Now, the JSON object involved here is the same in terms of values, but the properties are in a different order. I'm not sure if this is a result of the updated compiler or stdc++ version, updated boost lib, or something else moving to 18.04.

In short though, in this environment the tf test DOES pass. I'm not sure what you're seeing differently.

**Comment by beniz** [comment]
> Now, the JSON object involved here is the same in terms of values, but the properties are in a different order.

Thanks for the report, we will check again our tf builds to make sure.  Regarding the issue above, we've noticed it, try rebasing on master and it should go away, though we have only worked around it.

**Comment by beniz** [comment]
Fixed the broken TF tests on our side, was my fault, the model checkpoint had been incorrectly downloaded. So I do confirm the PR is fine, thanks for the good work! Feel free to switch the draft state off when readying for a merge.

**Comment by marcreichman** [comment]
Hi @beniz great news! I'm going to take draft off now. Merge when you wish.
